app.js

```diff
@@ -1,100 +1,74 @@
-document.addEventListener('DOMContentLoaded', () => {
-  // Selecting the necessary elements from the HTML file
-  const container = document.querySelector('.container');
-  const header = document.querySelector('.Pagetitle');
-  const form = document.querySelector('.form');
-  const bookDisplay = document.querySelector('.bookDisplay');
-  const inputTitle = document.querySelector('.title');
-  const inputAuthor = document.querySelector('.author');
-  const addButton = document.querySelector('.add');
-
-  // My Books constructor, for the user input
-  class Book {
-    constructor(title, author) {
-      this.title = title;
-      this.author = author;
-    }
+class DOMElements {
+  constructor() {
+    this.container = document.querySelector('.container');
+    this.header = document.querySelector('.Pagetitle');
+    this.form = document.querySelector('.form');
+    this.bookDisplay = document.querySelector('.bookDisplay');
+    this.inputTitle = document.querySelector('.title');
+    this.inputAuthor = document.querySelector('.author');
+    this.addButton = document.querySelector('.add');
+    this.error = document.createElement('span');
+    this.error.textContent = 'Please fill out all fields.';
   }
 
-  let bookHolder = [];
-
-  // Retrieving stored books from localStorage and displaying them
-  if (localStorage.getItem('storedBooks')) {
-    bookHolder = JSON.parse(localStorage.getItem('storedBooks'));
-    display();
+  displayError() {
+    this.bookDisplay.append(this.error);
   }
-
-  // Retrieving stored input data from localStorage and setting it to the input fields
-  const inputStorage = JSON.parse(localStorage.getItem('inputdata')) || {};
-  inputTitle.value = inputStorage.inputTitle || '';
-  inputAuthor.value = inputStorage.inputAuthor || '';
-
-  // Saving input data to localStorage whenever the input fields change
-  inputTitle.addEventListener('input', inputSave);
-  inputAuthor.addEventListener('input', inputSave);
-
-  function inputSave() {
-    inputStorage.inputTitle = inputTitle.value;
-    inputStorage.inputAuthor = inputAuthor.value;
-    localStorage.setItem('inputdata', JSON.stringify(inputStorage));
+}
+class LocalStorageHandler {
+  constructor() {
+    this.bookHolderKey = 'storedBooks';
+    this.inputDataKey = 'inputdata';
   }
 
-  // Adding a new book to the bookHolder array and updating localStorage and the UI
-  function addBook() {
-    const titleData = inputTitle.value;
-    const authorData = inputAuthor.value;
-    const newBook = new Book(titleData, authorData);
-    bookHolder.push(newBook);
-    localStorage.setItem('storedBooks', JSON.stringify(bookHolder));
-    inputTitle.value = '';
-    inputAuthor.value = '';
-    inputSave();
-    display();
+  getBookHolder() {
+    return JSON.parse(localStorage.getItem(this.bookHolderKey)) || [];
```

Comment:
- [ ]  You can deleted a create book, but you should use the method from your class. Like the way you did on saving books. please create a delete method in your classes and use it to delete the created book.  This will help you to master JavaScript OOP and understand more how JavaScript methods works.